bug: Docker build failing on `npm install pnpm`

I run into this error on the `RUN npm install pnpm` of the Dockerfile both when I attempt to run `docker build -t da-service .` locally as well as with VS Code Dev Containers:

```node
ERROR: failed to solve: process "/bin/sh -c npm install pnpm" did not complete successfully: exit code: 1
```

<details><summary>Details</summary>

```bash
[+] Building 72.0s (11/14)                                                                                                                                   
 => [internal] load .dockerignore                                                                                                                       0.0s
 => => transferring context: 123B                                                                                                                       0.0s
 => [internal] load build definition from Dockerfile                                                                                                    0.0s
 => => transferring dockerfile: 429B                                                                                                                    0.0s
 => [internal] load metadata for docker.io/library/node:18.14-alpine                                                                                    1.1s
 => [ 1/10] FROM docker.io/library/node:18.14-alpine@sha256:f8a51c36b0be7434bbf867d4a08decf0100e656203d893b9b0f8b1fe9e40daea                            0.0s
 => [internal] load build context                                                                                                                       0.0s
 => => transferring context: 383.18kB                                                                                                                   0.0s
 => CACHED [ 2/10] WORKDIR /usr/src/app                                                                                                                 0.0s
 => CACHED [ 3/10] COPY package*.json ./                                                                                                                0.0s
 => CACHED [ 4/10] COPY tsconfig.json ./                                                                                                                0.0s
 => CACHED [ 5/10] COPY src ./src                                                                                                                       0.0s
 => CACHED [ 6/10] COPY .env ./                                                                                                                         0.0s
 => ERROR [ 7/10] RUN npm install pnpm                                                                                                                 70.8s
------
 > [ 7/10] RUN npm install pnpm:
#0 62.30 npm WARN deprecated uuid@2.0.1: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
#0 62.61 npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
#0 68.46 npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
#0 70.12 npm notice 
#0 70.12 npm notice New minor version of npm available! 9.5.0 -> 9.6.5
#0 70.12 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.6.5>
#0 70.12 npm notice Run `npm install -g npm@9.6.5` to update!
#0 70.12 npm notice 
#0 70.12 npm ERR! code 1
#0 70.12 npm ERR! path /usr/src/app/node_modules/node-libcurl
#0 70.12 npm ERR! command failed
#0 70.12 npm ERR! command sh -c node-pre-gyp install --fallback-to-build
#0 70.12 npm ERR! Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/usr/src/app/node_modules/node-libcurl/lib/binding/node_libcurl.node --module_name=node_libcurl --module_path=/usr/src/app/node_modules/node-libcurl/lib/binding --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v108' (1)
#0 70.12 npm ERR! node-pre-gyp info it worked if it ends with ok
#0 70.12 npm ERR! node-pre-gyp info using node-pre-gyp@1.0.10
#0 70.12 npm ERR! node-pre-gyp info using node@18.14.2 | linux | arm64
#0 70.12 npm ERR! node-pre-gyp info check checked for "/usr/src/app/node_modules/node-libcurl/lib/binding/node_libcurl.node" (not found)
#0 70.12 npm ERR! node-pre-gyp http GET https://github.com/JCMais/node-libcurl/releases/download/v3.0.0/node_libcurl-v3.0.0-node-v108-linux-arm64-musl.tar.gz
#0 70.12 npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://github.com/JCMais/node-libcurl/releases/download/v3.0.0/node_libcurl-v3.0.0-node-v108-linux-arm64-musl.tar.gz 
#0 70.12 npm ERR! node-pre-gyp WARN Pre-built binaries not installable for node-libcurl@3.0.0 and node@18.14.2 (node-v108 ABI, musl) (falling back to source compile with node-gyp) 
#0 70.12 npm ERR! node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/JCMais/node-libcurl/releases/download/v3.0.0/node_libcurl-v3.0.0-node-v108-linux-arm64-musl.tar.gz 
#0 70.12 npm ERR! gyp info it worked if it ends with ok
#0 70.12 npm ERR! gyp info using node-gyp@9.3.1
#0 70.12 npm ERR! gyp info using node@18.14.2 | linux | arm64
#0 70.12 npm ERR! gyp info ok 
#0 70.12 npm ERR! gyp info it worked if it ends with ok
#0 70.12 npm ERR! gyp info using node-gyp@9.3.1
#0 70.12 npm ERR! gyp info using node@18.14.2 | linux | arm64
#0 70.12 npm ERR! gyp ERR! find Python 
#0 70.12 npm ERR! gyp ERR! find Python Python is not set from command line or npm configuration
#0 70.12 npm ERR! gyp ERR! find Python Python is not set from environment variable PYTHON
#0 70.12 npm ERR! gyp ERR! find Python checking if "python3" can be used
#0 70.12 npm ERR! gyp ERR! find Python - "python3" is not in PATH or produced an error
#0 70.12 npm ERR! gyp ERR! find Python checking if "python" can be used
#0 70.13 npm ERR! gyp ERR! find Python - "python" is not in PATH or produced an error
#0 70.13 npm ERR! gyp ERR! find Python 
#0 70.13 npm ERR! gyp ERR! find Python **********************************************************
#0 70.13 npm ERR! gyp ERR! find Python You need to install the latest version of Python.
#0 70.13 npm ERR! gyp ERR! find Python Node-gyp should be able to find and use Python. If not,
#0 70.13 npm ERR! gyp ERR! find Python you can try one of the following options:
#0 70.13 npm ERR! gyp ERR! find Python - Use the switch --python="/path/to/pythonexecutable"
#0 70.13 npm ERR! gyp ERR! find Python   (accepted by both node-gyp and npm)
#0 70.13 npm ERR! gyp ERR! find Python - Set the environment variable PYTHON
#0 70.13 npm ERR! gyp ERR! find Python - Set the npm configuration variable python:
#0 70.13 npm ERR! gyp ERR! find Python   npm config set python "/path/to/pythonexecutable"
#0 70.13 npm ERR! gyp ERR! find Python For more information consult the documentation at:
#0 70.13 npm ERR! gyp ERR! find Python https://github.com/nodejs/node-gyp#installation
#0 70.13 npm ERR! gyp ERR! find Python **********************************************************
#0 70.13 npm ERR! gyp ERR! find Python 
#0 70.13 npm ERR! gyp ERR! configure error 
#0 70.13 npm ERR! gyp ERR! stack Error: Could not find any Python installation to use
#0 70.13 npm ERR! gyp ERR! stack     at PythonFinder.fail (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:330:47)
#0 70.13 npm ERR! gyp ERR! stack     at PythonFinder.runChecks (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:159:21)
#0 70.13 npm ERR! gyp ERR! stack     at PythonFinder.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:202:16)
#0 70.13 npm ERR! gyp ERR! stack     at PythonFinder.execFileCallback (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/find-python.js:294:16)
#0 70.13 npm ERR! gyp ERR! stack     at exithandler (node:child_process:427:5)
#0 70.13 npm ERR! gyp ERR! stack     at ChildProcess.errorhandler (node:child_process:439:5)
#0 70.13 npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
#0 70.13 npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:289:12)
#0 70.13 npm ERR! gyp ERR! stack     at onErrorNT (node:internal/child_process:476:16)
#0 70.13 npm ERR! gyp ERR! stack     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
#0 70.13 npm ERR! gyp ERR! System Linux 5.15.49-linuxkit
#0 70.13 npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "configure" "--fallback-to-build" "--module=/usr/src/app/node_modules/node-libcurl/lib/binding/node_libcurl.node" "--module_name=node_libcurl" "--module_path=/usr/src/app/node_modules/node-libcurl/lib/binding" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v108"
#0 70.13 npm ERR! gyp ERR! cwd /usr/src/app/node_modules/node-libcurl
#0 70.13 npm ERR! gyp ERR! node -v v18.14.2
#0 70.13 npm ERR! gyp ERR! node-gyp -v v9.3.1
#0 70.13 npm ERR! gyp ERR! not ok 
#0 70.13 npm ERR! node-pre-gyp ERR! build error 
#0 70.13 npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node /usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js configure --fallback-to-build --module=/usr/src/app/node_modules/node-libcurl/lib/binding/node_libcurl.node --module_name=node_libcurl --module_path=/usr/src/app/node_modules/node-libcurl/lib/binding --napi_version=8 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v108' (1)
#0 70.13 npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/src/app/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
#0 70.13 npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
#0 70.13 npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
#0 70.13 npm ERR! node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
#0 70.13 npm ERR! node-pre-gyp ERR! System Linux 5.15.49-linuxkit
#0 70.13 npm ERR! node-pre-gyp ERR! command "/usr/local/bin/node" "/usr/src/app/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
#0 70.13 npm ERR! node-pre-gyp ERR! cwd /usr/src/app/node_modules/node-libcurl
#0 70.13 npm ERR! node-pre-gyp ERR! node -v v18.14.2
#0 70.13 npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.10
#0 70.13 npm ERR! node-pre-gyp ERR! not ok
#0 70.13 
#0 70.13 npm ERR! A complete log of this run can be found in:
#0 70.13 npm ERR!     /root/.npm/_logs/2023-04-29T11_25_18_036Z-debug-0.log
------
Dockerfile:11
--------------------
   9 |     
  10 |     # Install pnpm locally in your project directory
  11 | >>> RUN npm install pnpm
  12 |     
  13 |     # Install project dependencies using pnpm
--------------------
ERROR: failed to solve: process "/bin/sh -c npm install pnpm" did not complete successfully: exit code: 1

```

</details>

However, when I use 
```docker
FROM mcr.microsoft.com/devcontainers/typescript-node:18 as base
```
from [VS Code's Dev Container image repository](https://github.com/devcontainers/images/tree/main/src/typescript-node) at the top of the `Dockerfile` instead of 
```docker
FROM node:18.14-alpine as base
```
the build succeeds (both locally and in Dev Containers).

<details><summary>Details</summary>

```bash
[+] Building 180.1s (15/15) FINISHED                                                                                                   
 => [internal] load .dockerignore                                                                                                 0.0s
 => => transferring context: 123B                                                                                                 0.0s
 => [internal] load build definition from Dockerfile                                                                              0.0s
 => => transferring dockerfile: 462B                                                                                              0.0s
 => [internal] load metadata for mcr.microsoft.com/devcontainers/typescript-node:18                                               0.9s
 => [ 1/10] FROM mcr.microsoft.com/devcontainers/typescript-node:18@sha256:c258d970c416c860c973e1e8cd4b1c8e8245f187374dd15904ef  13.3s
 => => resolve mcr.microsoft.com/devcontainers/typescript-node:18@sha256:c258d970c416c860c973e1e8cd4b1c8e8245f187374dd15904ef1f4  0.0s
 => => sha256:0f398117670bb193f5c6cb1afff2f10d6904e5689050acb5cd7f21d6c7c73183 5.05kB / 5.05kB                                    0.0s
 => => sha256:80e7bb3929f7456a5e49b259d265f6c30093f52eea2f07aaf726ff784c9db628 22.34kB / 22.34kB                                  0.0s
 => => sha256:5884e7f2c8c61aa845de4902fc29639b58861ae6c2d80bafe82082c0456c0740 5.15MB / 5.15MB                                    1.4s
 => => sha256:dc80b8cdbfd36cb20231d807a50d704945d1df4da8f6e23197ccfcb629970491 53.71MB / 53.71MB                                  1.3s
 => => sha256:c258d970c416c860c973e1e8cd4b1c8e8245f187374dd15904ef1f4893da1069 1.61kB / 1.61kB                                    0.0s
 => => sha256:1b792b01ed3c8dc1488cd8aac41ab7d49bb17f3fa22b2e6c846078cec81a1c00 10.87MB / 10.87MB                                  0.3s
 => => sha256:8f993d5b17f32f4b8a535c25e182e5d8412625beee450e513ca57036e8fdd6dc 54.68MB / 54.68MB                                  1.2s
 => => sha256:c26edf98ece5babdb6b65e76738cef2c2d05e7075161753d3ba95cd84480d211 189.73MB / 189.73MB                                5.5s
 => => sha256:ef4553a29827be949323a49bbe126aac00893203a3234abe3a2915aa4b0b768a 4.20kB / 4.20kB                                    1.4s
 => => extracting sha256:dc80b8cdbfd36cb20231d807a50d704945d1df4da8f6e23197ccfcb629970491                                         1.3s
 => => sha256:14531ce9f0a46be86d89363abbf533499226c7734e56c3fbb7d70a8da9562be7 2.28MB / 2.28MB                                    1.8s
 => => sha256:f12d0fbf1676444f72f7a86719bd3c8a5ef70c77eaee331304d13da832785a54 45.58MB / 45.58MB                                  2.1s
 => => sha256:5f64eb351b344b535e5676014358c3045349fae2d10e4d45048067a26b17229c 452B / 452B                                        1.9s
 => => sha256:ab52160beb8c299b9cbc1871d4437a0d0078c195ada9dea92f2a53a82253e3c6 615B / 615B                                        2.0s
 => => sha256:e0f12e4dc616e9fec18d255bce2751e03ef5960f80ce27de2576ec82c7193135 152B / 152B                                        2.1s
 => => sha256:a33a868bf74e055264aa2e67907b2ee45baad5137175331a231f7124bc6bddce 6.26MB / 6.26MB                                    2.3s
 => => sha256:d9cb5ee1791cf6684e30ed87c27271dd350924a08a24aed403a26a8a0bce061c 424B / 424B                                        2.2s
 => => sha256:23c5221cc21eb8202f5b0db2749d236d2e521f59a415d6a0cc3d80bee7262403 140B / 140B                                        2.3s
 => => sha256:10036c11447f8cc70b7d8c0cc445c09566d0294f8dde91d4947e646fc418c1e0 226B / 226B                                        2.5s
 => => sha256:7c1be147ae7daee0b71b616dd01799d8fd2a6ea30a06a75ed75a6c9162e16162 242B / 242B                                        2.5s
 => => sha256:cc2dab73aa3b44c7df8ab5a98761184bc50e0ce81ce816def15ebbd45c6561dd 63.84MB / 63.84MB                                  3.5s
 => => sha256:b987a2df6707ff61a783e9a34937aa492905afe0ea58d46c30d30f4d410b790a 45.42MB / 45.42MB                                  5.3s
 => => extracting sha256:5884e7f2c8c61aa845de4902fc29639b58861ae6c2d80bafe82082c0456c0740                                         0.1s
 => => extracting sha256:1b792b01ed3c8dc1488cd8aac41ab7d49bb17f3fa22b2e6c846078cec81a1c00                                         0.2s
 => => extracting sha256:8f993d5b17f32f4b8a535c25e182e5d8412625beee450e513ca57036e8fdd6dc                                         1.2s
 => => sha256:56c4f704dae79d7c9189f368c3b25cf49a3468f2e52428fe90fd512a2cf60067 7.50MB / 7.50MB                                    3.8s
 => => sha256:4986921bf90f08cb3416b4091551f98b5877239ef0d6ecb4ea15e14f8ffd8576 45.01MB / 45.01MB                                  4.7s
 => => sha256:abb08069452b8612dde53d164d725d549da7499bf3c1a391c6cf00175fa79ecd 431B / 431B                                        4.8s
 => => sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1 32B / 32B                                          5.3s
 => => sha256:57ca398a74ea4a23446ebacabc36cc2f55617cf4cbc7b129cb2255984ebab8c2 231B / 231B                                        5.3s
 => => sha256:4a0a367cb9ab2a26403cf52f5c723ab43789ab0af63a8d46b3765d46c0f78e47 61.00MB / 61.00MB                                  6.7s
 => => sha256:36975edf19b3a66496b611274d51b2d10fd73084a97811fcc0e9b37b7d253562 247B / 247B                                        5.5s
 => => extracting sha256:c26edf98ece5babdb6b65e76738cef2c2d05e7075161753d3ba95cd84480d211                                         2.4s
 => => extracting sha256:ef4553a29827be949323a49bbe126aac00893203a3234abe3a2915aa4b0b768a                                         0.0s
 => => extracting sha256:f12d0fbf1676444f72f7a86719bd3c8a5ef70c77eaee331304d13da832785a54                                         0.7s
 => => extracting sha256:14531ce9f0a46be86d89363abbf533499226c7734e56c3fbb7d70a8da9562be7                                         0.1s
 => => extracting sha256:5f64eb351b344b535e5676014358c3045349fae2d10e4d45048067a26b17229c                                         0.0s
 => => extracting sha256:ab52160beb8c299b9cbc1871d4437a0d0078c195ada9dea92f2a53a82253e3c6                                         0.0s
 => => extracting sha256:e0f12e4dc616e9fec18d255bce2751e03ef5960f80ce27de2576ec82c7193135                                         0.0s
 => => extracting sha256:a33a868bf74e055264aa2e67907b2ee45baad5137175331a231f7124bc6bddce                                         0.2s
 => => extracting sha256:d9cb5ee1791cf6684e30ed87c27271dd350924a08a24aed403a26a8a0bce061c                                         0.0s
 => => extracting sha256:23c5221cc21eb8202f5b0db2749d236d2e521f59a415d6a0cc3d80bee7262403                                         0.0s
 => => extracting sha256:10036c11447f8cc70b7d8c0cc445c09566d0294f8dde91d4947e646fc418c1e0                                         0.0s
 => => extracting sha256:7c1be147ae7daee0b71b616dd01799d8fd2a6ea30a06a75ed75a6c9162e16162                                         0.0s
 => => extracting sha256:b987a2df6707ff61a783e9a34937aa492905afe0ea58d46c30d30f4d410b790a                                         0.6s
 => => extracting sha256:cc2dab73aa3b44c7df8ab5a98761184bc50e0ce81ce816def15ebbd45c6561dd                                         0.7s
 => => extracting sha256:56c4f704dae79d7c9189f368c3b25cf49a3468f2e52428fe90fd512a2cf60067                                         0.1s
 => => extracting sha256:4986921bf90f08cb3416b4091551f98b5877239ef0d6ecb4ea15e14f8ffd8576                                         0.7s
 => => extracting sha256:abb08069452b8612dde53d164d725d549da7499bf3c1a391c6cf00175fa79ecd                                         0.0s
 => => extracting sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1                                         0.0s
 => => extracting sha256:57ca398a74ea4a23446ebacabc36cc2f55617cf4cbc7b129cb2255984ebab8c2                                         0.0s
 => => extracting sha256:36975edf19b3a66496b611274d51b2d10fd73084a97811fcc0e9b37b7d253562                                         0.0s
 => => extracting sha256:4a0a367cb9ab2a26403cf52f5c723ab43789ab0af63a8d46b3765d46c0f78e47                                         0.7s
 => [internal] load build context                                                                                                 0.0s
 => => transferring context: 383.18kB                                                                                             0.0s
 => [ 2/10] WORKDIR /usr/src/app                                                                                                  1.3s
 => [ 3/10] COPY package*.json ./                                                                                                 0.0s
 => [ 4/10] COPY tsconfig.json ./                                                                                                 0.0s
 => [ 5/10] COPY src ./src                                                                                                        0.0s
 => [ 6/10] COPY .env ./                                                                                                          0.0s
 => [ 7/10] RUN npm install pnpm                                                                                                 81.6s
 => [ 8/10] RUN npx pnpm install                                                                                                 30.0s
 => [ 9/10] RUN chown -R node:node /usr/src/app                                                                                  37.1s
 => [10/10] RUN npm run build                                                                                                    11.1s
 => exporting to image                                                                                                            4.8s
 => => exporting layers                                                                                                           4.8s
 => => writing image sha256:4ad3cecb90c56f2139cbc66447a1c9eb07531b7d0e9257e3d7e614f4a13f9585                                      0.0s
 => => naming to docker.io/library/da-service
```

</details>

And then I can run `docker run -d -p 3008:3008 da-service` no problem.

**Environment**

Docker Desktop 4.19.0 on Apple M1 Pro using MacOS Ventura Version 13.3.1

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM node:18.14-alpine as base
+FROM mcr.microsoft.com/devcontainers/typescript-node:18 as base
 
 WORKDIR /usr/src/app
 

